src/pt/lycantoons/src/eu/kanade/tachiyomi/extension/pt/lycantoons/LycanToons.kt

```diff
@@ -0,0 +1,228 @@
+
+    override fun popularMangaParse(response: Response): MangasPage =
+        response.parseAs<PopularResponse>().data.toMangasPage()
+
+    // =====================Latest=====================
+
+    override fun latestUpdatesRequest(page: Int): Request = metricsRequest("recently-updated", page)
+
+    override fun latestUpdatesParse(response: Response): MangasPage =
+        response.parseAs<PopularResponse>().data.toMangasPage()
+
+    // =====================Search=====================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val payload = SearchRequestBody(
+            limit = PAGE_LIMIT,
+            page = page,
+            search = query,
+            seriesType = filters.valueOrEmpty<SeriesTypeFilter>(),
+            status = filters.valueOrEmpty<StatusFilter>(),
+            tags = filters.selectedTags(),
+        )
+
+        val body = json.encodeToString(payload).toRequestBody(JSON_MEDIA_TYPE)
+        return POST("$baseUrl/api/series", headers, body)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage =
+        response.parseAs<SearchResponse>().series.toMangasPage()
+
+    override fun getFilterList(): FilterList = FilterList(
+        SeriesTypeFilter(),
+        StatusFilter(),
+        TagsFilter(),
+    )
+
+    // =====================Details=====================
+
+    override fun mangaDetailsRequest(manga: SManga): Request = seriesRequest(manga.slug())
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val result = response.parseAs<SeriesDto>()
+        return result.toSManga().apply { initialized = true }
+    }
+
+    // =====================Chapters=====================
+
+    override fun chapterListRequest(manga: SManga): Request = seriesRequest(manga.slug())
+
+    override fun chapterListParse(response: Response): List<SChapter> =
+        response.parseAs<SeriesDto>().let { series ->
+            series.capitulos.orEmpty()
+                .map { it.toSChapter(series.slug) }
+                .sortedByDescending { it.chapter_number }
+        }
+
+    // =====================Pages========================
+
+    override fun pageListRequest(chapter: SChapter): Request = GET(
+        "$baseUrl${chapter.url}",
+        headers.newBuilder()
+            .add("Referer", "$baseUrl/")
+            .build(),
+    )
+
+    override fun pageListParse(response: Response): List<Page> {
+        val html = response.body.string()
+        val (slug, chapterNumber) = response.extractSlugAndChapter()
+
+        val pageCount = pageCountFromHtml(html, chapterNumber)
+            ?.takeIf { it > 0 }
+            ?: error("Quantidade de páginas não encontrada no HTML para capítulo '$chapterNumber'")
+
+        val chapterPath = "$cdnUrl/$slug/$chapterNumber"
+        return List(pageCount) { index ->
+            val imageUrl = "$chapterPath/page-$index.jpg"
+            Page(index, imageUrl, imageUrl)
```

Comment:
```suggestion
            Page(index, imageUrl = imageUrl)
```

---

Or if you actually need the page the image is rendered from, you would so something like (making a lot of assumptions here):
```suggestion
            Page(index, response.request.url.toString(), imageUrl)
```